src/yaca_conv.rs

```diff
@@ -0,0 +1,296 @@
+const YACA_KEY_TYPE_EC_PARAMS:  c_int = 13;
+
+
+pub(crate) fn key_type_rs_to_c(kt: &KeyType) -> c_int
+{
+    match kt {
+        KeyType::Symmetric => YACA_KEY_TYPE_SYMMETRIC,
+        KeyType::Des => YACA_KEY_TYPE_DES,
+        KeyType::Iv => YACA_KEY_TYPE_IV,
+        KeyType::Rsa(KeySubType::Public) => YACA_KEY_TYPE_RSA_PUB,
+        KeyType::Rsa(KeySubType::Private) => YACA_KEY_TYPE_RSA_PRIV,
+        KeyType::Dsa(KeySubType::Public) => YACA_KEY_TYPE_DSA_PUB,
+        KeyType::Dsa(KeySubType::Private) => YACA_KEY_TYPE_DSA_PRIV,
+        KeyType::Dh(KeySubType::Public) => YACA_KEY_TYPE_DH_PUB,
+        KeyType::Dh(KeySubType::Private) => YACA_KEY_TYPE_DH_PRIV,
+        KeyType::Ec(KeySubType::Public) => YACA_KEY_TYPE_EC_PUB,
+        KeyType::Ec(KeySubType::Private) => YACA_KEY_TYPE_EC_PRIV,
+        KeyType::Rsa(KeySubType::Params) => panic!("No KeySubType::Params for RSA"),
+        KeyType::Dsa(KeySubType::Params) => YACA_KEY_TYPE_DSA_PARAMS,
+        KeyType::Dh(KeySubType::Params) => YACA_KEY_TYPE_DH_PARAMS,
+        KeyType::Ec(KeySubType::Params) => YACA_KEY_TYPE_EC_PARAMS,
+    }
+}
+
+pub(crate) fn key_type_c_to_rs(kt: c_int) -> KeyType
+{
+    match kt {
+        YACA_KEY_TYPE_SYMMETRIC => KeyType::Symmetric,
+        YACA_KEY_TYPE_DES => KeyType::Des,
+        YACA_KEY_TYPE_IV => KeyType::Iv,
+        YACA_KEY_TYPE_RSA_PRIV => KeyType::Rsa(KeySubType::Private),
+        YACA_KEY_TYPE_RSA_PUB => KeyType::Rsa(KeySubType::Public),
+        YACA_KEY_TYPE_DSA_PRIV => KeyType::Dsa(KeySubType::Private),
+        YACA_KEY_TYPE_DSA_PUB => KeyType::Dsa(KeySubType::Public),
+        YACA_KEY_TYPE_DSA_PARAMS => KeyType::Dsa(KeySubType::Params),
+        YACA_KEY_TYPE_EC_PRIV => KeyType::Ec(KeySubType::Private),
+        YACA_KEY_TYPE_EC_PUB => KeyType::Ec(KeySubType::Public),
+        YACA_KEY_TYPE_EC_PARAMS => KeyType::Ec(KeySubType::Params),
+        YACA_KEY_TYPE_DH_PRIV => KeyType::Dh(KeySubType::Private),
+        YACA_KEY_TYPE_DH_PUB => KeyType::Dh(KeySubType::Public),
+        YACA_KEY_TYPE_DH_PARAMS => KeyType::Dh(KeySubType::Params),
+        x => panic!("Wrong key_type passed from C: {}", x),
+    }
+}
+
+const YACA_KEY_LENGTH_EC_PRIME192V1: c_int = 0x300000C0;
+const YACA_KEY_LENGTH_EC_PRIME256V1: c_int = 0x30000100;
+const YACA_KEY_LENGTH_EC_SECP256K1:  c_int = 0x31200100;
+const YACA_KEY_LENGTH_EC_SECP384R1:  c_int = 0x31100180;
+const YACA_KEY_LENGTH_EC_SECP521R1:  c_int = 0x31100209;
+
+const YACA_KEY_LENGTH_DH_RFC_1024_160: c_int = 0x20000400;
+const YACA_KEY_LENGTH_DH_RFC_2048_224: c_int = 0x21000800;
+const YACA_KEY_LENGTH_DH_RFC_2048_256: c_int = 0x22000800;
+
+const YACA_KEY_LENGTH_DH_GENERATOR_2: c_int = 0x10000000;
+const YACA_KEY_LENGTH_DH_GENERATOR_5: c_int = 0x11000000;
+
+pub(crate) fn key_length_rs_to_c(kl: &KeyLength) -> size_t
+{
+    match kl {
+        KeyLength::Bits(bl) => *bl as size_t,
+        KeyLength::Ec(KeyLengthEc::Prime192V1) => YACA_KEY_LENGTH_EC_PRIME192V1 as size_t,
```

Comment:
```-D unused-qualifications``` shows this qualification (and a bunch of others) to be unnecessary. I know it's explicitly qualified on purpose but wanted to point it out regardless.

Reply:
$ cargo rustc --lib -- -D unused-qualifications >/dev/null 2>&1; echo $?
0